src/Npgsql/Replication/PgOutput/PgOutputReplicationOptions.cs

```diff
@@ -15,11 +15,11 @@ public class PgOutputReplicationOptions : IEquatable<PgOutputReplicationOptions>
     /// <param name="publicationName">The publication names to include into the stream</param>
     /// <param name="protocolVersion">The version of the logical streaming replication protocol</param>
     /// <param name="binary">Send values in binary representation</param>
-    /// <param name="streaming">Enable streaming of in-progress transactions</param>
+    /// <param name="streamingMode">Enable streaming of in-progress transactions</param>
     /// <param name="messages">Write logical decoding messages into the replication stream</param>
     /// <param name="twoPhase">Enable streaming of prepared transactions</param>
-    public PgOutputReplicationOptions(string publicationName, ulong protocolVersion, bool? binary = null, bool? streaming = null, bool? messages = null, bool? twoPhase = null)
-        : this(new List<string> { publicationName ?? throw new ArgumentNullException(nameof(publicationName)) }, protocolVersion, binary, streaming, messages, twoPhase)
+    public PgOutputReplicationOptions(string publicationName, PgOutputProtocolVersion protocolVersion, bool? binary = null, PgOutputStreamingMode? streamingMode = null, bool? messages = null, bool? twoPhase = null)
```

Comment:
I'm not sure making the protocol version an enum warrants the breaking change here - the stream mode is clearly no longer a bool, but the protocol continues to just be an incrementing number (and it's very hard to imagine additional "options" to the protocol). So maybe it's better to just keep it a ulong as before...

If we really feel like it, we can also keep the old signature with [Obsolete], calling the new one (to start being a bit more backwards-compat friendly - there's little downside to it in this case).

Reply:
I agree that it's not important.
What warrants it IMO is the fact that we can restrict the range to values we actually support + consumers get autocomplete and documentation comments that give low-threshold information about when you actually need that version.
I was actually thinking about also throwing informative Exceptions from PgOutputReplicationOptions when you enable a feature that isn't supported by the protocol version you chose or alternatively making the protocol version optional and infer the minimum required version from the options you enabled when it isn't set, but that would be a little bit agaiinst the spirit of providing a very thin layer over the protocol.

Reply:
OK. Yeah, I'd definitely stay with being very explicit and up-front about the protocol version, rather than any fancy inferring.

Sure, I can see the value of an enum. Just in the interest of starting to make the replication API a bit more backwards-compatible, how about adding back the previous signature (with a numeric protocol number), and putting [Obsolete] on it, pointing to the new one? I know some people are building libraries/stuff on top of this, and this would make it easier ot move forward.